rust/azure_iot_operations_connector/src/file_mount_azure_device_registry.rs

```diff
@@ -1,4 +1,270 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-//! Wrapper for Azure Device Registry Client that uses file mount to get names and create/delete notifications.
+//!Azure Device Registry Client that uses file mount to get names and create/delete notifications.
+
+use notify::RecommendedWatcher;
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use tokio::sync::mpsc::UnboundedReceiver;
+use tokio::sync::oneshot;
+
+const ADR_RESOURCES_NAME_MOUNT_PATH: &str = "/etc/akri/config/adr_resources_names";
+
+/// A client that interacts with the file mount
+///
+/// This client provides functionality to retrieve device names and handle
+/// create/delete notifications from the Azure Device Registry.
+#[allow(dead_code)]
+pub struct FileMountClient {
+    /// The path to the file mount used by the client.
+    pub mount_path: PathBuf,
```

Comment:
if these don't need to be accessed by the user of the FileMountClient, make them not `pub`